### ISSUE TITLE
replace `mixed` with new `unknown` type?

### ISSUE THREAD

**Issue by mmkal** [gcanti/io-ts#204]
Since TypeScript 3.0, there's a new type called `unknown`. From the announcement:

```typescript
let foo: unknown = 10;

// Since `foo` has type `unknown`, TypeScript
// errors on each of these locations.
foo.x.prop;
foo.y.prop;
foo.z.prop;
foo();
new foo();
upperCase(foo);
foo `hello world!`;

function upperCase(x: string) {
    return x.toUpperCase();
}
```

It seems like this is _very_ similar to `mixed` but since it's native, would it make sense to drop `mixed` in favour of `unknown`?

**Comment by gcanti** [comment]
@mmkal looks like the community agrees on considering this a breaking change, see this discussion https://github.com/gcanti/fp-ts/issues/539

Personally I think that a good balance in order to avoid stagnation would be to support last `n` stable releases of TypeScript (when `n` is to be decided).

**Comment by mmkal** [comment]
Makes sense. It is technically a breaking change, unless there's some way to shim `unknown` for older typescript versions. Not aware of anything though.

Unrelated - what do you think about setting up a https://gitter.im community for io-ts (which could also probably be used for fp-ts, io-ts-codegen etc.) for this sort of thing? It's not exactly an 'issue', which means it's unlikely to get much discussion here. But it'd be a good place to share ideas on novel ways to use the system, check if certain things have been done before etc.

**Comment by sledorze** [comment]
So good :)
